torch/distributions/constraints.py

```diff
@@ -185,21 +187,46 @@ class _LowerTriangular(Constraint):
     Constrain to lower-triangular square matrices.
     """
     def check(self, value):
-        return (torch.tril(value) == value).min(-1)[0].min(-1)[0]
+        masked_value = value * torch.tril(value.new(*value.shape[-2:]).fill_(1.0))
```

Comment:
I now see the problem with 3D inputs, but can we please make batch tril a helper in utils that:
1. just calls tril on 2D inputs
2. does this workaround with multiplication otherwise
I'd rather do it that way, because the workaround makes it slower in the 2d case